staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/interface.go

```diff
@@ -21,9 +21,22 @@ import (
 	"time"
 )
 
-// QueueSetFactory is used to create QueueSet objects.
+// QueueSetFactory is used to create QueueSet objects.  Creation, like
+// config update, is done in two phases: the first phase consumes the
+// QueuingConfig and the second consumes the DispatchingConfig.  They
+// are separated so that errors from the first phase can be found
+// before committing to a concurrency allotment for the second.
 type QueueSetFactory interface {
-	NewQueueSet(config QueueSetConfig) (QueueSet, error)
+	// QualifyQueuingConfig does the first phase of creating a QueueSet
+	QualifyQueuingConfig(QueuingConfig) (QueueSetCompleter, error)
```

Comment:
I can live with this name, but I'd prefer one that sounded more build-y. `BeginConstruction`?

Reply:
Would you still say that if you knew it did not actually build anything?
I can see how we might want something that is complementary to "complete", as a matter of having the words in the interface make sense.